cmd/kubeadm/app/util/apiclient/wait_test.go

```diff
@@ -30,39 +31,42 @@ func TestGetControlPlaneComponents(t *testing.T) {
 		expected []controlPlaneComponent
 	}{
 		{
-			name: "port values from config",
+			name: "port and addresses from config",
 			cfg: &kubeadmapi.ClusterConfiguration{
 				APIServer: kubeadmapi.APIServer{
 					ControlPlaneComponent: kubeadmapi.ControlPlaneComponent{
 						ExtraArgs: []kubeadmapi.Arg{
 							{Name: "secure-port", Value: "1111"},
+							{Name: "bind-address", Value: "0.0.0.0"},
 						},
 					},
 				},
 				ControllerManager: kubeadmapi.ControlPlaneComponent{
 					ExtraArgs: []kubeadmapi.Arg{
 						{Name: "secure-port", Value: "2222"},
+						{Name: "bind-address", Value: "0.0.0.0"},
 					},
 				},
 				Scheduler: kubeadmapi.ControlPlaneComponent{
 					ExtraArgs: []kubeadmapi.Arg{
 						{Name: "secure-port", Value: "3333"},
+						{Name: "bind-address", Value: "0.0.0.0"},
 					},
 				},
 			},
 			expected: []controlPlaneComponent{
-				{name: "kube-apiserver", url: "https://127.0.0.1:1111/healthz"},
-				{name: "kube-controller-manager", url: "https://127.0.0.1:2222/healthz"},
-				{name: "kube-scheduler", url: "https://127.0.0.1:3333/healthz"},
+				{name: "kube-apiserver", url: fmt.Sprintf("https://0.0.0.0:1111/%s", endpointLivez)},
```

Comment:
/hold

this is wrong, 0.0.0.0 is not a real address, it means listen in all existing address on the host

Reply:
0.0.0.0 is the default value for all --bind-address flags and it would return `ok` on some setups as in ` curl https://0.0.0.0:6443/livez`
but regardless, i updated the unit tests to have scoped addresses.

Reply:
I mean, 0.0.0.0 is not a real address, it has an special meaning on sockets to listen in all available address on host. https://en.wikipedia.org/wiki/0.0.0.0

`curl https://0.0.0.0:6443` does not do what you may think it does, it works but it is because linux interpret it as

> In Linux a program may specify 0.0.0.0 as the remote address to connect to the current host (AKA [localhost]

but that is an imlementation detail of linux,

Reply:
from the networking perspective, I recommend that , if users set 0.0.0.0 as bind address, you direct it to 127.0.0.1